Added stage and local inputs to plan JSONs

pavelvelikhov opened the pull request.

pavelvelikhov committed:
Added stage and local inputs to plan JSONs

pavelvelikhov committed:
Added stage and local inputs to plan JSONs

dcherednik committed:
Add grpc request proxy wilson span and trace id on the request context. (#606)

VPolka committed:
KIKIMR-20081: add grant/revoke permissions to query service (#522)

* add grant/revore permissions to query service

* fix rebase

* fix issues

domwst committed:
KIKIMR-20533 Implemented conversion from OTEL header to traceId (#607)

* KIKIMR-20533 Implemented conversion from OTEL header to traceId

nshestakov committed:
Replace Y_ABORT_UNLESS #2 (#603)

domwst committed:
KIKIMR-20533 TraceId parser unittests (#617)

SammyVimes committed:
KIKIMR-20217 Initial datashard tracing (#530)

vitstn committed:
init (#610)

nepal committed:
Fix ui.sh, add missing mounts.txt (#614)

* Fix ui.sh, add missing mounts.txt
* mounts.txt for uig.sh, drop no longer needed LD_LIBRARY_PATH

pavelvelikhov committed:
Intermediate commit for debugging

pavelvelikhov committed:
Intermediate commit for debugging

Enjection committed:
Add dynamic config cli (#615)

* add dynamic config cli

---------

Co-authored-by: Innokentii Mokin <innokentii@ydb.tech>

mxkovalev committed:
YQL-17250: Add operation information into hybrid statistics (#597)

* YQL-17250: Add operation information into hybrid statistics

* Fix review comments

rvu1024 committed:
[dq] Integrate YT peephole transforms into DQ integration (YQL-17386) (#605)

azevaykin committed:
Out<NYql::TIssues> (#626)

alexv-smirnov committed:
Import libs 1 (#590)

* Import libs 1

* Add new file without extension

* Add file missed in export config

dcherednik committed:
Use user provided otel header to start grpc proxy tracing. (#628)

pavelvelikhov committed:
Fixed the problems with inputs

pavelvelikhov committed:
Fixed the problems with inputs

Gazizonoki committed:
Moved csv parsing in import file cmd to YDB CLI and supported pg-types (only row tables) (#514)

* Fixed parsing JSON from pg types

* Fixed bugs

* Moved csv parsing in import file cmd to YDB CLI and supported pg-types (only row tables)

* Fixed bugs

* Fixed bug

* Fixed bugs

* Fixed test

* Added column table validation

* Fixed bugs

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -836,19 +852,31 @@ class TxPlanSerializer {
 
-                    Visit(inputCn.Output().Stage(), inputPlanNode);
+                    auto planNode = Visit(inputCn.Output().Stage(), inputPlanNode);
+                    inputIds.emplace_back(planNode);
```
вот здесь в `inputIds` правильнее  добавить `inputPlanNode`

ulya-sidorina replied:
и тогда можно вернуть прежнюю сигнатуру этой функции

StekPerepolnen committed:
added vscode (#636)

eivanov89 committed:
add talks/presentations to the docs (#528)

* add talks/presentations to the docs

* Update ydb/docs/en/core/public-talks.md

---------

Co-authored-by: Ivan Blinkov <ivan@blinkov.ru>

StekPerepolnen committed:
KIKIMR-20321 viewer tests asan fix (#525)

KIKIMR-20231/asan-fix

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -824,8 +842,6 @@ class TxPlanSerializer {
             YQL_ENSURE(stagePlanNode.StageProto, "Could not find a stage with id " << stageGuid);
             SerializerCtx.StageGuidToId[stageGuid] = SerializerCtx.PlanNodeId;
             VisitedStages.insert(expr.Raw());
-            auto node = expr.Cast<TDqStageBase>().Program().Body().Ptr();
```
а почему пришлось поменять порядок обхода ?

resetius committed:
First 25 queries of TPC-DS (#632)

azevaykin committed:
CheckWriteUnit & FinishProposeWriteUnit (#631)

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -1042,13 +1088,15 @@ class TxPlanSerializer {
         } else {
-            return TMaybe<ui32>();
+            auto inputs = Visit(toflow.Input().Ptr(), planNode);
+            Y_ENSURE(inputs.size()==1);
```
так лучше не делать, Visit может вернуть пустой результат

ulya-sidorina replied:
на самом деле ветку `else` можно вообще удалить

resetius committed:
Add schema example for TPC-DS (#613)

ijon committed:
Move iam auth actors out of IO pool (#648)

Move non-blocking iam auth actors out of IO pool so that blocking logging actors couldn't prevent them from working by blocking entire pool. Also IO pool is not a suitable place for a non-blocking actor.

KIKIMR-19644

vitstn committed:
fix (#654)

nepal committed:
[YQL-16903] Introduce BlockEngine pragma (parser/config provider part) (#616)

* initial

* add tests

resetius committed:
Second 25 queries of TPC-DS (#660)

resetius committed:
Third 25 queries of TPC-DS (#662)

dorooleg committed:
cgi parameters have been sorted YQ-2558 (#619)

rvu1024 committed:
[yql] Fix join-anyjoin_common_dup test (#651)

azevaykin committed:
UpdateRow & WriteRow (#663)

Tony-Romanov committed:
Fix Unique/Distinct constraints from nested filtering FlatMap. (#584)

* Add KDevelop

* Fix Unique/Distinct constraints from nested filtering FlatMap.

* A little fix.

Commit:
YQ Connector: fix integration tests style (#640)

rvu1024 committed:
[dq] Separate setting SplitStageOnDqReplicate (YQL-17382) (#664)

kungasc committed:
KIKIMR-19521 BTreeIndex Test (#653)

kungasc committed:
KIKIMR-19139 Bugfix load indexes (#657)

kruall committed:
add StarvingInRowForNotEnoughCpu for tcp sessions (#625)

Co-authored-by: Aleksandr Kriukov <kruall@ydb.ru>

kruall committed:
Fix metric PotentialMaxThreadCount (#629)

Co-authored-by: Aleksandr Kriukov <kruall@ydb.ru>

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -174,7 +174,7 @@ class TxPlanSerializer {
     {}
 
-    void Serialize() {
+    void Serialize() {    
```
лишние пробелы

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -265,6 +265,7 @@ class TxPlanSerializer {
     };
 
     void WritePlanNodeToJson(const TQueryPlanNode& planNode, NJsonWriter::TBuf& writer) const {
+
```
лишняя пустая строка

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -836,19 +852,31 @@ class TxPlanSerializer {
                 }
             }
 
+            for (const auto& arg : expr.Cast<TDqStageBase>().Program().Args()) {
```
лучше убедиться , что `inputIds.size() == expr.Cast<TDqStageBase>().Program().Args().size()`

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -300,17 +301,34 @@ class TxPlanSerializer {
-                    writer.EndList();
+                    if (std::holds_alternative<ui32>(input)) {
+                        writer.BeginObject();
+                        writer.WriteKey("Local");
```
давай более понятное название, например InternalOperatorId/ExternalPlanNodeId

resetius committed:
Last 25 queries of TPC-DS (#668)

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -936,44 +967,59 @@ class TxPlanSerializer {
         } else if (auto maybeDelete = TMaybeNode<TKqpDeleteRows>(node)) {
             operatorId = Visit(maybeDelete.Cast(), planNode);
+        } else if (auto maybeArg = TMaybeNode<TCoArgument>(node)) {
+            operatorId = Visit(maybeArg.Cast(), planNode);
```
тут просто можно `return {node};` даже визитом писать не нужно

EgorkaZ committed:
YQ-2628: add connections option for streaming queries (#579)

kungasc committed:
KIKIMR-19521 BTreeIndex Bugfix index size (#671)

azevaykin committed:
WriteRow out of space (#665)

pavelvelikhov committed:
Fixed Yulia's comments

pavelvelikhov committed:
Fixed Yulia's comments

ulya-sidorina commented on ydb/core/kqp/opt/kqp_query_plan.cpp:
```diff
@@ -300,17 +300,34 @@ class TxPlanSerializer {
-                    writer.EndList();
+                    if (std::holds_alternative<ui32>(input)) {
+                        writer.BeginObject();
+                        writer.WriteKey("Local");
```
вот тут еще переименовать

pavelvelikhov committed:
Minor change in JSON output

pavelvelikhov committed:
Minor change in JSON output

Darych committed:
YQL-17087: Add channel spilling to dq pipe communication (#612)

* Add channel spilling to dq pipe communication
* Add EvPoison handler to DqChannelStorageActor

robdrynkin committed:
KIKIMR-20521: Add UseVDisksBalancing feature flag (#523)

Co-authored-by: robdrynkin <robdrynkin@nebius.com>

robdrynkin committed:
KIKIMR-20522: Tests for vdisks balancing (#531)

Co-authored-by: robdrynkin <robdrynkin@nebius.com>

alexv-smirnov committed:
Library import 2 (#639)

eivanov89 committed:
Revert "Workload TPC-C init" (#529)

This reverts commit da025ccc23fafcc3a3a572d83f82996be433fa89.

alexv-smirnov committed:
Consistent debug (#652)

mxkovalev committed:
YQL-17250: Add operation name in fallback message (#627)

* YQL-17250: Add operation name in fallback message

* Fix review comments

domwst committed:
Moved TSpan::Link(TraceId) implementation to .cpp (#674)

rvu1024 committed:
[yql] Fix result diff in dq_file tests (YQL-17386, YQL-17404, YQL-17402, YQL-17405) (#676)

serbel324 committed:
Add BsCostTracker and advanced cost metrics, KIKIMR-17759 (#600)

* Add BsCostTracker

* Remove redundant Y_UNUSDE

* Remove redundant HugeBlobSize redefinition

* Add estimation for huge writes

alexv-smirnov committed:
Import libs 3 (#679)

StekPerepolnen committed:
added tests (#647)

nshestakov committed:
Refactoring - optimize parameters of CreatePartitionWriter (#669)

vitstn committed:
YQL-15941 avoid llvm version pinning where possible (#677)

* init

* build

Hor911 committed:
Ignore empty metrics in in-progress stats. Try-catch in-progress stats conversion. (#633)

kungasc committed:
KIKIMR-19521 BTreeIndex Fix SchemeShard Verify (#655)

jepett0 committed:
CREATE / DROP VIEW queries support (#656)

* CREATE / DROP VIEW queries support

SELECTing from VIEWs is not supported yet!

- Make changes to the YQL grammar to add CREATE / DROP VIEW statements
- Save the query text specified in the CREATE VIEW statement in the VIEW object
- Implement VIEW as a wrapped generic Object: CREATE / DROP VIEW
commands call Create/DropObject commands with appropriate settings
- Query text is stored in the object's features

* Implement placeholder intent determination transformer for CreateObject to allow type annotation of the select statement saved inside a VIEW in CREATE VIEW statement

Saving parsed AST of the view can be used to find type errors in the select statement saved inside a VIEW during validation stage of the YQL pipeline. It seems to necessitate adding this placeholder implementation of the CreateObject handler

CyberROFL committed:
Mute postgresql* tests (#675)

domwst committed:
KIKIMR-20042 Fix of TSpan abort outside of actor system (#630)

* KIKIMR-20042 Fix of TSpan abort outside of actor system

* Removed explicit comparsion with nullptr

basil-cow committed:
[YQL-17338] disable cast warning from integral types to float types (#690)

Tony-Romanov committed:
Fix constrains after expand PartitionsByKeys. (#685)

* Fix constrains after expand PartitionsByKeys.

* A little fix and cosmetic.

domwst committed:
KIKIMR-20533 Forward traceparent header to KV-tablet (#689)

* KIKIMR-20533 Forward traceparent header to KV-tablet

* Introduced TWilsonGrpc::RequestActor

resetius committed:
Add TPC-DS Q56 (#692)

EgorkaZ committed:
YQ-2628: Print stats on final status log (#563)

pavelvelikhov committed:
Merge branch 'KIKIMR-20577' of github.com:pavelvelikhov/ydb into KIKIMR-20577
